File: 234-palindrome-linked-list/234-palindrome-linked-list.cs
public class Solution {
    public bool IsPalindrome(ListNode head) {        
        var s = head;
        var f = head;
        
        while(f!=null && f.next!=null){
            s=s.next;
            f=f.next.next;
        }
        
        var a = reverse(s);
        
        while(a!=null && head!=null){
            if(a.val!=head.val)return false;
            a=a.next;
            head=head.next;
        }
        return true;
    }
    public ListNode reverse(ListNode head){
        ListNode prev= null;
        while(head!=null){
            var next = head.next;
            head.next=prev;
            prev=head;
            head=next;
        }
        return prev;
    }
}

File: 234-palindrome-linked-list/234-palindrome-linked-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    bool isPalindrome(ListNode* head) {
        ListNode *mid = head;
        ListNode *temp = head;
        while(temp && temp->next){
            temp=temp->next;
            temp=temp->next;
            mid=mid->next;
        }
        ListNode *rev = reverse(mid);
        while(rev && head){
            if(rev->val!=head->val){
                return false;
            }
            rev=rev->next;
            head=head->next;
        }
        return true;
    }
    ListNode* reverse(ListNode* head){
        ListNode* prev=NULL;
        while(head){
            ListNode *next=head->next;
            head->next=prev;
            prev=head;
            head=next;
        }
        return prev;
    }
};